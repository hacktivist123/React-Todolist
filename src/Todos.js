import React from 'react';

const Todos = ({ todos, deleteTodo }) => {

    const todoList = todos.length !== 0 ? (
        todos.map( (todo, i) => {
            return(
                <div className= "collection-item" key={i}>
                    <span onClick={ () => {deleteTodo(todo.id)}}>{todo.content}</span>
                    
                </div>
            )
        })
    ) : (
        <p className="center">You have No todo's left, Yay!</p>
    )
    return (
        <div className="todos collection">
        { todoList }

        </div>
    )
}

export default Todos;