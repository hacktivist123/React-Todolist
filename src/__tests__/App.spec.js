import React from 'react'
import { shallow, mount } from 'enzyme'
import App from '../App'
import Todos from '../Todos'

describe('Main Page Test', () => {
  it('should render a h1 element correctly', () => {
  const wrapper = shallow(<App deleteTodo={jest.fn()} addTodo={jest.fn()}/>)
  const text = wrapper.find('h1')
  expect(text.text()).toBe("Todos")
  
  });

  it('should addTodo to state', () => {
    const wrapper = mount(<App />);

    wrapper.setState({content: 'test'})
    wrapper.find('button').simulate('click', {
      preventDefault: () => {}
    })

    expect(wrapper.state().content).toBe('test')

  })

  it('should delete todo', () => {
    const deleteTodo=jest.fn()

  const wrapper = mount(<Todos todos={['test']} deleteTodo={deleteTodo}/>);
    wrapper.find('span').simulate('click', {});

    expect(wrapper).toMatchSnapshot();
  })

});