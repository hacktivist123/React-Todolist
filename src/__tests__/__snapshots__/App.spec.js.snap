// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page Test should delete todo 1`] = `
<Todos
  deleteTodo={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  todos={
    Array [
      "test",
    ]
  }
>
  <div
    className="todos collection"
  >
    <div
      className="collection-item"
      key="0"
    >
      <span
        onClick={[Function]}
      />
    </div>
  </div>
</Todos>
`;
